tests/test_qnn.py

```diff
@@ -0,0 +1,429 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for the pennylane.qnn module.
+"""
+import os
+import tempfile
+
+import numpy as np
+import pytest
+
+try:
+    import tensorflow as tf
+    from tensorflow.keras.layers import Layer
+    from tensorflow.keras.initializers import RandomNormal
+
+    CORRECT_TF_VERSION = int(tf.__version__.split(".")[0]) > 1
+except ImportError:
+    from abc import ABC
+
+    Layer = ABC
+    CORRECT_TF_VERSION = False
+
+import pennylane as qml
+from pennylane.qnn import KerasLayer
+
+min_tf = pytest.mark.skipif(
+    not CORRECT_TF_VERSION, reason="TensorFlow version 2 and above required"
+)
```

Comment:
There is also
```python
tf = pytest.importorskip("tensorflow", minversion="2.0")
```
(from here: http://doc.pytest.org/en/latest/skipping.html#skipping-on-a-missing-import-dependency)

I think it might be a new feature!

Reply:
Also: when you use `skipif` as a module level variable, it will be executed on the module level, so acts globally on the whole file. So I don't think you need to use it as a decorator below (but not 100% sure)